File: April 16, 2026.js
function doMath(str) {

  let lastNumber = null;
  let number = "";
  let charCount = 0;
  let result = 0;
  let start = false;

  for (let i = 0; i < str.length; i++) {

    if (str[i] >= "0" && str[i] <= "9") {
      number += str[i];
      start = true;
    } else {

       if (number !== "") {
        const num = Number(number);

        if (lastNumber === null) {
          result = num;
        } else {
          if (charCount % 2 === 0) {
            result += num;
          } else {
            result -= num;
          }
        }

        lastNumber = num;
        number = "";
        charCount = 0;
      }

      if (start) {
        charCount += 1;
      }
    }
  }

  if (number !== "") {
    const num = Number(number);

    if (lastNumber === null) {
      result = num;
    } else {
      if (charCount % 2 === 0) {
        result += num;
      } else {
        result -= num;
      }
    }
  }

  return result;
}

File: April 16, 2026.py
def do_math(s):

    last_number = None
    number = ""
    char_count = 0
    result = 0
    start = False

    for char in s:
        if char.isdigit():
            number += char
            start = True
        else:
            if number != "":
                num = int(number)
                if last_number == None:
                    result = num
                else:
                    if char_count % 2 == 0:
                        result += num
                    else:
                        result -= num
            
                last_number = num;
                number = ""
                char_count = 0
            
            if start:
                char_count += 1

    if number != "":
        num = int(number)
        if last_number == None:
            result = num
        else:
            if char_count % 2 == 0:
                    result += num
            else:
                result -= num

    return result

do_math("3ab10c8")